File: tournaments/bfsDistancesUnweightedGraph/bfsDistancesUnweightedGraph.py
def bfsDistancesUnweightedGraph(matrix, startVertex):
    from collections import deque

    n = len(matrix)
    queue = deque([startVertex])
    result = [-1] * n
    result[startVertex] = 0

    while queue:
        current = queue.pop()

        for nextVertex in range(n):
            if -1 == result[nextVertex] and matrix[current][nextVertex]:
                queue.appendleft(nextVertex)
                result[nextVertex] = 1 + result[current]

    return result

File: tournaments/bfsDistancesUnweightedGraph/bfsDistancesUnweightedGraph.js
function bfsDistancesUnweightedGraph(matrix, startVertex) {
    const queue = [startVertex];
    const result = Array(matrix.length).fill(-1);
    
    result[startVertex] = 0;
    
    while (queue.length) {
        const current = queue.shift();
        
        for (let nextVertex = 0; nextVertex in matrix; ++nextVertex) {
            if (!~result[nextVertex] && matrix[current][nextVertex]) {
                queue.unshift(nextVertex);
                result[nextVertex] = 1 + result[current];
            }
        }
    }
    
    return result;
}
